common/ferrostar/src/navigation_controller/models.rs

```diff
@@ -2,6 +2,19 @@ use crate::deviation_detection::{RouteDeviation, RouteDeviationTracking};
 use crate::models::{RouteStep, SpokenInstruction, UserLocation, VisualInstruction, Waypoint};
 use geo::LineString;
 
+/// A subset of state values that are used to show the user their current progress along the full route/trip.
+#[derive(Debug, Clone, PartialEq, uniffi::Record)]
+pub struct ArrivalState {
```

Comment:
Yeah, I agree with `TripProgress`. I had this debate with myself while building it all out and ended up deciding that including `distanceToNextManeuver` was logical. Aside from general organizational reasoning, it was also beneficial from a code standpoint since it was technically part of the same calculations that ultimately build trip progress.